spec/requests/posts_specs.rb

```diff
@@ -1,32 +1,42 @@
 require 'rails_helper'
 
-RSpec.describe 'Posts', type: :request do
-  context 'the index method' do
-    it 'show all users' do
-      get('/users/10/posts')
-      expect(assigns(:posts)).to eq(Post.all)
+RSpec.describe Post, type: :model do
+  describe 'For the Post model' do
+    before(:each) do
+      @user = User.new(name: 'Tom', photo: 'image.png', bio: 'Teacher from Mexico', post_counter: 0)
+      @post = Post.new(author: @user, title: 'Test', text: 'testing', likes_counter: 7, comments_counter: 5)
     end
 
-    it 'render index template' do
-      get('/users/10/posts')
-      expect(response).to render_template('index')
+    before { @post.save }
+
+    it 'if there is title' do
+      @post.title = true
+      expect(@post).to be_valid
     end
 
-    it 'the veiw body is correct (index)' do
-      get('/users/10/posts')
-      expect(response.body).to include('Index for user posts')
+    it 'if there is max 250 characters' do
+      @post.title = 'Testing'
+      expect(@post).to be_valid
+    end
+
+    it 'if likes counter is integer' do
+      @post.likes_counter = 5
+      expect(@post).to be_valid
+    end
+
+    it 'if likes counter greater than or equal to zero' do
+      @post.likes_counter = -9
+      expect(@post).to_not be_valid
     end
-  end
 
-  context 'the show method' do
-    it 'show specific user post veiw' do
-      get('/users/403/posts/1')
-      expect(response.body).to include('specific user post')
+    it 'if comments counter greater than or equal to zero.' do
+      @post.comments_counter = -5
+      expect(@post).to_not be_valid
     end
 
-    it 'render the correct page' do
-      get('/users/403/posts/1')
-      expect(response.body).to render_template('show')
+    it 'if comments counter is integer' do
+      @post.comments_counter = 8
+      expect(@post).to be_valid
     end
   end
 end
```

Comment:
- [ ] Good job adding some tests but please kindly include integration tests for `Post` index and show page with the following requirements.

### Index Page
* Tests that asserts you can see the user's profile picture.
* Tests that asserts you can see the user's username.
* Tests that asserts you can see the number of posts the user has written.
* Tests that asserts you can see a post's title.
* Tests that asserts you can see some of the post's body.
* Tests that asserts you can see the first comments on a post.
* Tests that asserts you can see how many comments a post has.
* Tests that asserts you can see how many likes a post has.
* Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
* Tests that redirects to the post's show page when you click on a post.

### Show Page
* Tests that asserts you can see the post's title.
* Tests that asserts you can see who wrote the post.
* Tests that asserts you can see how many comments it has.
* Tests that asserts you can see how many likes it has.
* Tests that asserts you can see the post body.
* Tests that asserts you can see the username of each commentor.
* Tests that asserts you can see the comment each commentor left.